File: drivers/php/src/Helpers/Querier.php
<?php
/**
 * Minimal contract helpers need. {@see \Reddb\Conn} satisfies it via
 * {@see Conn::query()}; tests pass fakes that record SQL.
 */

declare(strict_types=1);

namespace Reddb\Helpers;

interface Querier
{
    /**
     * Run a SQL query with positional `$N` parameters. Returns the raw
     * JSON envelope as a string (mirrors {@see \Reddb\Conn::query()}).
     *
     * @param array<int,mixed> $params
     */
    public function query(string $sql, array $params = []): string;
}

File: drivers/java/src/main/java/dev/reddb/helpers/Querier.java
package dev.reddb.helpers;

/**
 * Minimal contract helpers need. {@link dev.reddb.Conn} satisfies it via
 * {@code query(String, Object...)}; tests pass fakes that record SQL.
 */
@FunctionalInterface
public interface Querier {
    /** Run a SQL query with positional {@code $N} parameters. Returns raw JSON envelope. */
    byte[] query(String sql, Object... params);
}
